source/slang/slang-compiler.h

```diff
@@ -1132,7 +1132,10 @@ namespace Slang
         SlangTargetFlags    targetFlags = 0;
         Slang::Profile      targetProfile = Slang::Profile();
         FloatingPointMode   floatingPointMode = FloatingPointMode::Default;
-        bool                isWholeProgramRequest = false;
+        bool isWholeProgramRequest()
+        {
+            return (targetFlags & SLANG_TARGET_FLAG_GENERATE_WHOLE_PROGRAM) != 0;
+        }
```

Comment:
We might eventually need/want to have certain targets that *always* generate a whole-program output. CPU and CUDA targets would seem like obvious candidates for this, but SPIR-V and DXIL might also qualify if we can work through any complication.

For formats and targets that support multiple entry points, we already have the rule that the client of Slang needs to look up entry points by name in the output blob we produce (at least, I think that is the case...), so we could always return a multiple-entry-point blob in response to a request for a single entry point.

I guess that defaulting to such behavior might confuse users, but I also don't like having to make users explicitly opt into what is obviously the right behavior for targets like CPU/CUDA...

Reply:
Actually, I think what we *might* want to do here is to have the public API operations for getting a whole-program result take an (optional) output parameter for a blob of diagnostics, and then they can get `getOrCreateWholeProgramResult` instead of `getExistingWholeProgramResult`, which would mean that they could be used to query the whole-program result *and* generate it on demand if it wasn't already created. That approach would mean that even if an application leaves out the `SLANG_TARGET_FLAG_GENERATE_WHOLE_PROGRAM` flag, they can still query the whole-program result.